spec/migrations/20191023180330_migrate_ems_ref_obj_to_ems_ref_type_spec.rb

```diff
@@ -0,0 +1,115 @@
+        :type        => "ManageIQ::Providers::Vmware::InfraManager::Vm",
+        :ems_ref     => "vm-123",
+        :ems_ref_obj => <<~EMS_REF_OBJ
+          --- !ruby/string:VimString
+          str: vm-1234
+          xsiType: :ManagedObjectReference
+          vimType: :VirtualMachine
+        EMS_REF_OBJ
+      )
+      host = host_stub.create!(
+        :type        => "ManageIQ::Providers::Vmware::InfraManager::HostEsx",
+        :ems_ref     => "host-123",
+        :ems_ref_obj => <<~EMS_REF_OBJ
+          --- !ruby/string:VimString
+          str: host-1234
+          xsiType: :ManagedObjectReference
+          vimType: :HostSystem
+        EMS_REF_OBJ
+      )
+      cluster = cluster_stub.create!(
+        :ems_ref     => "domain-c1234",
+        :ems_ref_obj => <<~EMS_REF_OBJ
+          --- !ruby/string:VimString
+          str: domain-c1234
+          xsiType: :ManagedObjectReference
+          vimType: :ClusterComputeResource
+        EMS_REF_OBJ
+      )
+      respool = resource_pool_stub.create!(
+        :ems_ref     => "resgroup-1234",
+        :ems_ref_obj => <<~EMS_REF_OBJ
+          --- !ruby/string:VimString
+          str: resgroup-1234
+          xsiType: :ManagedObjectReference
+          vimType: :ResourcePool
+        EMS_REF_OBJ
+      )
+      storage = storage_stub.create!(
+        :ems_ref     => "datastore-1234",
+        :ems_ref_obj => <<~EMS_REF_OBJ
+          --- !ruby/string:VimString
+          str: datastore-1234
+          xsiType: :ManagedObjectReference
+          vimType: :Datastore
+        EMS_REF_OBJ
+      )
+
+      migrate
+
+      expect(vm.reload.ems_ref_type).to      eq("VirtualMachine")
+      expect(host.reload.ems_ref_type).to    eq("HostSystem")
+      expect(cluster.reload.ems_ref_type).to eq("ClusterComputeResource")
+      expect(respool.reload.ems_ref_type).to eq("ResourcePool")
+      expect(storage.reload.ems_ref_type).to eq("Datastore")
+    end
+
+    it "ignores ems_ref_obj if it isn't a VimString" do
+      vm = vm_stub.create!(
+        :type        => "ManageIQ::Providers::Redhat::InfraManager::Vm",
+        :ems_ref     => "b7212ae9-e968-4431-bb17-cc16d5095cd0",
+        :ems_ref_obj => "--- b7212ae9-e968-4431-bb17-cc16d5095cd0\n"
+      )
+
+      migrate
+
+      expect(vm.reload.ems_ref_type).to be_nil
+    end
+  end
+
+  migration_context :down do
+    it "migrates an ems_ref_type to an ems_ref_obj" do
+      vm = vm_stub.create!(:ems_ref => "vm-1234", :ems_ref_type => "VirtualMachine")
+      migrate
+      expect(vm.reload.ems_ref_obj).to eq <<~EMS_REF_OBJ
+        --- !ruby/string:VimString
+        str: vm-1234
+        xsiType: :ManagedObjectReference
+        vimType: :VirtualMachine
+      EMS_REF_OBJ
+    end
+
+    it "sets to a simple yaml string if ems_ref_type is nil" do
+      vm = vm_stub.create!(:ems_ref => "b7212ae9-e968-4431-bb17-cc16d5095cd0")
+      migrate
+      expect(vm.reload.ems_ref_obj).to eq("--- b7212ae9-e968-4431-bb17-cc16d5095cd0\n")
```

Comment:
Did this happen for all providers or only specific ones?

Reply:
Good point, cloud vms wouldn't have ems_ref_obj set.  I'll skip those

Reply:
Okay skipping cloud vms now